types/hapi/index.d.ts

```diff
@@ -12,2712 +13,58 @@
-     *  * 'public' - mark the response as suitable for public caching.
-     *  * 'private' - mark the response as suitable only for private caching.
-     */
-    privacy?: 'default' | 'public' | 'private';
-    /** an array of HTTP response status codes (e.g. 200) which are allowed to include a valid caching directive. Defaults to [200]. */
-    statuses?: number[];
-    /** a string with the value of the 'Cache-Control' header when caching is disabled. Defaults to 'no-cache'. */
-    otherwise?: string;
-} & ({
-    /** relative expiration expressed in the number of milliseconds since the item was saved in the cache. Cannot be used together with expiresAt. */
-    expiresIn?: number;
-    /** time of day expressed in 24h notation using the 'HH:MM' format, at which point all cache records for the route expire. Cannot be used together with expiresIn. */
-    expiresAt?: undefined;
-} | {
-    /** relative expiration expressed in the number of milliseconds since the item was saved in the cache. Cannot be used together with expiresAt. */
-    expiresIn?: undefined;
-    /** time of day expressed in 24h notation using the 'HH:MM' format, at which point all cache records for the route expire. Cannot be used together with expiresIn. */
-    expiresAt?: string;
-} | {
-    /** relative expiration expressed in the number of milliseconds since the item was saved in the cache. Cannot be used together with expiresAt. */
-    expiresIn?: undefined;
-    /** time of day expressed in 24h notation using the 'HH:MM' format, at which point all cache records for the route expire. Cannot be used together with expiresIn. */
-    expiresAt?: undefined;
-});
-
-/**
- * For context see RouteAdditionalConfigurationOptions > cors
- */
-export interface CorsConfigurationObject {
-    /** a strings array of allowed origin servers ('Access-Control-Allow-Origin'). The array can contain any combination of fully qualified origins along with origin strings containing a wildcard '*' character, or a single '*' origin string. Defaults to any origin ['*']. */
-    origin?: string[] | '*';
-    /** number of seconds the browser should cache the CORS response ('Access-Control-Max-Age'). The greater the value, the longer it will take before the browser checks for changes in policy. Defaults to 86400 (one day). */
-    maxAge?: number;
-    /** a strings array of allowed headers ('Access-Control-Allow-Headers'). Defaults to ['Accept', 'Authorization', 'Content-Type', 'If-None-Match'] */
-    headers?: string[];
-    /** a strings array of additional headers to headers. Use this to keep the default headers in place. */
-    additionalHeaders?: string[];
-    /** a strings array of exposed headers ('Access-Control-Expose-Headers'). Defaults to ['WWW-Authenticate', 'Server-Authorization']. */
-    exposedHeaders?: string[];
-    /** a strings array of additional headers to exposedHeaders. Use this to keep the default headers in p
```

Comment:
why have the additional folder `definitions`? all these files are definitions by definition :)

Reply:
@mhegazy At first, with @BorntraegerMarc, we thought that should be more organized and easy to mantain  many small files separated by functionality (server, request, response, lifecycle, etc) instead of one file with some thousand of lines. But of course I can change if the recommendation is one single file.

Reply:
Agree with @rafaelsouzaf -> better folder structure = smaller files = easier readable code = more maintainability

Reply:
That still doesn't answer why the files are in `hapi/definitions` and not just in `hapi`.